torch/_inductor/codegen/wrapper.py

```diff
@@ -415,3 +494,112 @@ def call_kernel(self, name: str, kernel: Kernel):
+            output_refs = [x.codegen_reference() for x in V.graph.graph_outputs]
+            if output_refs:
+                if len(output_refs) == 1:
+                    output_types = "at::Tensor"
+                else:
+                    output_return_type = "at::Tensor"
+                    output_return_types = [output_return_type] * len(output_refs)
+                    output_return_types = ", ".join(output_return_types)
+                    output_types = f"std::tuple<{output_return_types}>"
+            else:
+                output_types = "void"
+
+            if inp_len != 0:
+                inputs_args = ["at::Tensor"] * len(V.graph.graph_inputs.keys())
+                inputs_args = ", ".join(inputs_args)
+                inputs_args = f"std::tuple<{inputs_args}>"
+
+                self.prefix.writeline(
+                    f"{output_types} call_{self._call_func_id}({inputs_args} args) {{"
+                )
+                inputs_keys_str = ", ".join(V.graph.graph_inputs.keys())
+                self.prefix.writeline(f"at::Tensor {inputs_keys_str};")
+                self.prefix.writeline(f"std::tie({inputs_keys_str}) = args;")
+            else:
+                self.prefix.writeline(
+                    f"{output_types} call_{self._call_func_id}(std::tuple<> args) {{"
+                )
+            for name in V.graph.randomness_seeds:
+                self.prefix.writeline(f"at::Tensor {name};")
+                self.prefix.writeline(
+                    f"{name} = at::randint(std::pow(2, 31), {{}}, at::ScalarType::Long);"
+                )
+            V.graph.sizevars.codegen(self.prefix, V.graph.graph_inputs)
+
+    def write_allocate_line(self, buffer):
+        self.writeline(CppAllocateLine(buffer))
+
+    def write_del_line(self, name):
+        return
+
+    def write_free_if_not_reused_line(self, buffer):
+        return
+
+    def write_reuse_line(self, input_buffer, output_buffer):
+        self.writeline(CppReuseLine(input_buffer, output_buffer))
+
+    def get_deferred_line(self, name, layout):
+        return DeferredLine(
+            name, f"auto {name} = {layout.view.codegen_reference()};  // alias"
+        )
+
+    def generate_return(self, result, output_refs):
+        if output_refs:
+            if len(output_refs) == 1:
+                result.writeline("return " + output_refs[0] + "; }''' )")
+            else:
+                result.writeline(
+                    "return std::make_tuple(" + ", ".join(output_refs) + "); }''' )"
+                )
+        else:
+            result.writeline("return; }''' )")
+
+    def generate_end(self, result):
+        result.splice(
+            f"""
+            module = load_inline(
+                name='inline_extension',
+                cpp_sources=[wrapper],
+                functions=['call_{self._call_func_id}'],
+                extra_cflags=['-DCPU_CAPABILITY_AVX2 -march=native -O3 -ffast-math -fno-finite-math-only -fopenmp'])
```

Comment:
Add a call into codecache here to get the args used.  Some platforms (e.g. M1 mac) may need different flags.

Reply:
Updated the code to get the cpp compile cmd from codecache.